datafusion/core/src/physical_plan/metrics/mod.rs

```diff
@@ -161,8 +161,8 @@ impl Metric {
     }
 
     /// return a reference to the partition
-    pub fn partition(&self) -> &Option<usize> {
-        &self.partition
+    pub fn partition(&self) -> Option<&usize> {
+        self.partition.as_ref()
     }
```

Comment:
```suggestion
    pub fn partition(&self) -> Option<usize> {
        self.partition.copied()
     }
```

Reply:
I use `.clone()` instead of `.copied()` or `.cloned()`, because the compiler will raise error in last case:

```
error[E0599]: `Option<usize>` is not an iterator
   --> datafusion\core\src\physical_plan\metrics\mod.rs:165:24
    |
165 |         self.partition.copied()
    |                        ^^^^^^ `Option<usize>` is not an iterator
    |
   ::: C:\Users\gaoxinge\.rustup\toolchains\nightly-x86_64-pc-windows-gnu\lib/rustlib/src/rust\library\core\src\option.rs:518:1
    |
518 | pub enum Option<T> {
    | ------------------ doesn't satisfy `Option<usize>: Iterator`
    |
    = note: the following trait bounds were not satisfied:
            `Option<usize>: Iterator`
            which is required by `&mut Option<usize>: Iterator`
```

This error may refer to [Strange .cloned() error message mentioning iterators](https://github.com/rust-lang/rust/issues/74121).

Reply:
Aah yes, got the various methods confused :+1:

Reply:
I remove the `.clone()` to pass the ci, because it will cause [the clippy error](https://rust-lang.github.io/rust-clippy/master/index.html#clone_on_copy):

```
error: using `clone` on type `std::option::Option<usize>` which implements the `Copy` trait
   --> datafusion/core/src/physical_plan/metrics/mod.rs:165:9
    |
165 |         self.partition.clone()
    |         ^^^^^^^^^^^^^^^^^^^^^^ help: try removing the `clone` call: `self.partition`
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#clone_on_copy
    = note: `-D clippy::clone-on-copy` implied by `-D warnings`
```